app/services/preview_service.rb

```diff
@@ -1,72 +1,61 @@
 # frozen_string_literal: true
 
 class PreviewService
-  attr_reader :document
+  attr_reader :edition
 
-  def initialize(document)
-    @document = document
+  def initialize(edition)
+    @edition = edition
   end
 
-  def create_preview(args)
-    save_changes(args)
-    publish_draft(document)
+  def create_preview
+    upload_assets(edition)
+    publish_draft(edition)
   end
 
-  def try_create_preview(args)
-    save_changes(args)
-    try_publish_draft(document) unless has_issues?
+  def try_create_preview
+    if has_issues?
+      edition.update!(revision_synced: false)
+    else
+      begin
+        upload_assets(edition)
```

Comment:
Just a small one - can we just call `create_preview` here instead of repeating these lines?

Reply:
Yup - good catch

Reply:
@emmabeynon refactored in 5a4d96cdbc4088848ffb97db4123b67e8f50debe